dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/AbstractCommandExecutor.java

```diff
@@ -249,6 +261,22 @@ public TaskResponse run(String execCommand, TaskCallBack taskCallBack) throws Ex
             }
         }
 
+        if (podLogOutputFuture != null) {
+            try {
+                // Wait kubernetes pod log collection finished
+                String msg = (String) podLogOutputFuture.get();
+                if (StringUtils.isEmpty(msg)) {
+                    // delete pod after successful execution
+                    ProcessUtils.cancelApplication(taskRequest);
```

Comment:
Why need to cancel when pod log output is empty? Does it necessarily mean that the pod is finished?

Reply:
The pod is deleted in `cancelApplication` only when task runs successfully and we collect whole logs.